tests/integration/sync/test_sync_build_in_source.py

```diff
@@ -0,0 +1,72 @@
+import os
+import pytest
+import logging
+import json
+from unittest import skip
+from parameterized import parameterized
+
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+from tests.testing_utils import run_command_with_input
+from samcli.lib.utils.resources import AWS_LAMBDA_FUNCTION
+
+LOG = logging.getLogger(__name__)
+
+
+@skip("Building in source option is not exposed yet. Stop skipping once it is.")
+class TestSyncInfra_BuildInSource_Makefile(SyncIntegBase):
+    dependency_layer = False
+
+    def setUp(self):
+        super().setUp()
+        self.new_files_in_source = [
+            self.test_data_path.joinpath(
+                "code/before/makefile_function_create_new_file/file-created-from-makefile-function"
+            ),
+            self.test_data_path.joinpath("code/before/makefile_layer_create_new_file/file-created-from-makefile-layer"),
+        ]
+
+    def tearDown(self):
+        super().tearDown()
+        for path in self.new_files_in_source:
+            if os.path.isfile(path):
+                os.remove(path)
+
+    @parameterized.expand(
+        [
+            (True, True),  # build in source
+            (False, False),  # don't build in source
+            (None, False),  # use default for workflow (don't build in source)
+        ]
+    )
+    def test_sync_builds_and_deploys_successfully(self, build_in_source, new_file_should_be_in_source):
```

Comment:
@mndeveci do you think tests with code, watch, ADL are also necessary? I was thinking they probably shared the same building logic, but let me know what you think.

Reply:
Good point, let's add them as parameterized, we don't want to miss an edge case there. You can check some existing ones, they should have parameters on top of class definition.

Reply:
done, added sync --code tests